src/templates/modules/FM_Template_v1.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+// Internal
+import {IOrchestrator_v1} from
+    "src/orchestrator/interfaces/IOrchestrator_v1.sol";
+import {ERC165Upgradeable, Module_v1} from "src/modules/base/Module_v1.sol";
+import {IFM_Template_v1} from "./IFM_Template_v1.sol";
+import {IFundingManager_v1} from "@fm/IFundingManager_v1.sol";
+
+// External Libraries
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title   Inverter Template Funding Manager
+ *
+ * @notice  Basic template funding manager used as base for developing new
+ *          funding managers.
+ *
+ * @dev     This contract is used to showcase a basic setup for a funding
+ *          manager. The contract showcases the following:
+ *          - Inherit from the Module_v1 contract to enable interaction with
+ *            the Inverter workflow.
+ *          - Use of the IFundingManager_v1 interface to facilitate
+ *            interaction as a Funding Manager.
+ *          - Implement custom interface which has all the public facing
+ *            functions, errors, events and structs.
+ *          - Pre-defined layout for all contract functions, modifiers, state
+ *            variables etc.
+ *          - Use of the ERC165Upgradeable contract to check for interface
+ *            support.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer
+ *                          to our Security Policy at security.inverter.network
+ *                          or email us directly!
+ *
+ * @custom:version 1.0.0
+ *
+ * @author  Inverter Network
+ */
+contract FM_Template_v1 is IFM_Template_v1, Module_v1 {
+    //--------------------------------------------------------------------------
+    // Libraries
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // ERC165
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId_ == type(IFM_Template_v1).interfaceId
+            || interfaceId_ == type(IFundingManager_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+
+    //--------------------------------------------------------------------------
+    // Constants
+
+    // Add constants here
+
+    //--------------------------------------------------------------------------
+    // State
+
+    /// @dev    Mapping of user addresses to their deposited token amounts.
+    mapping(address => uint) internal _depositedAmounts;
```

Comment:
We want to name mappings so it's immediately clear what a mapping does internally. For example here that could be:

```suggestion
    mapping(address user => uint amount) internal _depositedAmounts;
```

Reply:
Updated